ruby/lib/minitest/queue/build_status_recorder.rb

```diff
@@ -52,6 +52,7 @@ def record(test)
         if (test.failure || test.error?) && !test.skipped?
           build.record_error("#{test.klass}##{test.name}", dump(test), stats: stats)
         else
+          build.record_flaky("#{test.klass}##{test.name}", stats: stats)
```

Comment:
My main concern is that we double the number of Redis calls we do for a successful run.